### PR TITLE
[CI][Bugfix] Fix 500 errors from priority overflow and TemplateError subclasses in schema fuzz tests

### PR DESCRIPTION
**Pull request by AndreasKaratzas**
The `test_openapi_stateless` schemathesis fuzz test is failing on AMD CI ([BuildKite #6544, entrypoints server api 1, mi325 and mi355](https://buildkite.com/vllm/amd-ci/builds/6544/steps/canvas?sid=019cf015-1cfd-445d-b66c-88374d4dc1cb&tab=output)). Schemathesis generates schema-valid inputs that trigger 500s from the server. This PR fixes three independent issues found during that investigation.

1. `priority` field has no int64 bounds

`priority` is defined as a plain `int` with no constraints. Values outside `[-2^63, 2^63 - 1]` cause `msgspec.msgpack` to raise `OverflowError` when serializing the request for the engine. That error gets swallowed by the broad `except Exception` in `AsyncLLM.generate()`, re-raised as `EngineGenerateError()` with no message, and the client gets back `{"message": "", "type": "InternalServerError"}`. The fix is to validate at the HTTP boundary with Pydantic `ge`/`le` constraints — the same approach already used for `seed` and `truncate_prompt_tokens`.

2. jinja2 `TemplateError` subclasses incorrectly return 500

`create_error_response` in `entrypoints/utils.py` identified jinja2 errors by checking `exc.__class__.__name__ == "TemplateError"`. That only matches the base class. Subclasses like `UndefinedError` and `TemplateRuntimeError` fall through to the default `else` branch and come back as 500 instead of 400.

3. Bare `assert` produces an empty 500

`assert final_res is not None` in `chat_completion_full_generator` raises `AssertionError` with no message if the generator yields nothing, which again produces `{"message": ""}`. Replaced with an explicit error response.

### Changes

Priority bounds: adds `ge=-(2**63), le=2**63-1` to the `priority` field across all five endpoint protocol files: `chat_completion`, `completion`, `responses`, `disagg`, and `pooling`.

Torch cleanup: while adding those bounds, also follows through on the TODO left in #34468: removes `import torch` and `_LONG_INFO = torch.iinfo(torch.long)` from the three protocol files that had it (`chat_completion`, `completion`, `responses`). These are pure Pydantic files and torch doesn't belong there. All usages are replaced with plain `_INT64_MIN = -(2**63)` / `_INT64_MAX = 2**63 - 1` constants.

Error handling: fixes the `TemplateError` subclass check in `utils.py` with an MRO walk, and replaces the bare assert in `chat_completion/serving.py` with a proper error response.

### Related

- #34468: int64 bounds on `DetokenizeRequest.tokens`; introduced the `torch.iinfo` TODO
- #35231: `ge=0` bounds on `CompletionRequest.prompt` token IDs

### Test plan

- `pytest tests/entrypoints/openai/test_openai_schema.py -v -s`
- `pytest tests/entrypoints/openai/chat_completion/test_chat_error.py -v -s`

cc @kenroche

### CODE REVIEW

File: vllm/entrypoints/pooling/base/protocol.py [hunk 1]
```diff
@@ -34,6 +34,8 @@ class PoolingBasicRequestMixin(OpenAIBaseModel):
     )
     priority: int = Field(
         default=0,
+        ge=-(2**63),
+        le=2**63 - 1,
```
[the remark this answers is not in the document]
Comment (AndreasKaratzas) [reply vllm/entrypoints/pooling/base/protocol.py:38]: This file only has the one priority field using these bounds, so a named constant would be defined and used exactly once. -(2**63) / 2**63 - 1 are self-documenting. The named constants in the other files (e.g. completion/protocol.py) are justified because they're reused across seed, truncate_prompt_tokens, and priority.

File: vllm/entrypoints/serve/disagg/protocol.py [hunk 2]
```diff
@@ -93,6 +93,8 @@ def validate_token_ids(cls, v: list[int]) -> list[int]:
     )
     priority: int = Field(
         default=0,
+        ge=-(2**63),
+        le=2**63 - 1,
```
[the remark this answers is not in the document]
Comment (AndreasKaratzas) [reply vllm/entrypoints/serve/disagg/protocol.py:97]: Same reasoning as the pooling comment above. Also worth noting this is consistent with #34468, which used inline 2**63 - 1 for the same reason.

**Review by DarkLight1337 — APPROVED** [review]

**Comment by AndreasKaratzas** [comment]
@DarkLight1337 Sry, had to add missing dependency from:
- https://github.com/vllm-project/vllm/pull/36775
